File: apps/modernization-api/src/main/java/gov/cdc/nbs/patient/profile/race/PatientRace.java
package gov.cdc.nbs.patient.profile.race;

import java.time.LocalDate;
import java.util.Collection;

record PatientRace(
    long patient,
    long id,
    short version,
    LocalDate asOf,
    Race category,
    Collection<Race> detailed
) {

  record Race(
      String id,
      String description
  ) {
  }
}

File: apps/modernization-ui/src/pages/patient/profile/race/PatientRace.ts
export type PatientRace = {
    __typename?: 'PatientRace';
    patient: number;
    version: number;
    asOf: any;
    category: { __typename?: 'PatientCodedValue'; id: string; description: string };
    detailed?: Array<{ __typename?: 'PatientCodedValue'; id: string; description: string }>;
};
